cfme/infrastructure/virtual_machines.py

```diff
@@ -1666,10 +1694,27 @@ def step(self, *args, **kwargs):
 
 @navigator.register(InfraTemplateCollection, 'TemplatesOnly')
 class TemplatesAll(CFMENavigateStep):
-    VIEW = TemplatesOnlyAllView
     prerequisite = NavigateToSibling('All')
 
+    @property
+    def VIEW(self):  # noqa
+        from cfme.infrastructure.provider import ProviderTemplatesView
+        from cfme.infrastructure.provider import InfraProvider
+        try:
+            if self.obj.parent.one_of(InfraProvider):
+                return ProviderTemplatesView  # ProviderTemplatesOnlyAllView?? What is this?
+        except Exception:
+            pass
+        return TemplatesOnlyAllView
+
     def step(self, *args, **kwargs):
+        from cfme.infrastructure.provider import InfraProvider
```

Comment:
Can we move this to the top of the file?

Reply:
As mentioned above, InfraProvider will cause import errors. I don't love having these imports but I feel it is a better check than just seeing if a parent exists. I have seen checks for just a parent in other places, but wanted to check for InfraProvider.